### PR TITLE
feat: Add exit status code to finch

### PR DESCRIPTION
**Pull request by Shubhranshu153**
Issue #, if available:

*Description of changes:*
 add exit status code to finch  to return exit code if interface has exit codes.

*Testing done:*
shubhum@147ddaa42911 finch % finch run --rm alpine sh -c "exit 10"
shubhum@147ddaa42911 finch % echo $?
10

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/finch/main.go [hunk 1]
```diff
@@ -32,6 +33,7 @@ func main() {
 	mem := fmemory.NewMemory()
 	stdOut := os.Stdout
 	if err := xmain(logger, stdLib, fs, stdLib, mem, stdOut); err != nil {
+		errutil.HandleExitCoder(err)
```
Comment (swagatbora90) [inline cmd/finch/main.go:36]: Can we add a unit test for this?

Comment (Shubhranshu153) [reply cmd/finch/main.go:36]: the package is from nerdctl, are we saying adding unit test for the Handle exit coder ?

Comment (swagatbora90) [reply cmd/finch/main.go:36]: Not for the exit coder, but for the behavior change.

Comment (Shubhranshu153) [reply cmd/finch/main.go:36]: may be integ test, dont see the point of adding unit test to main() function
Can add a test that runs a container and exits and we can see the status code. (similar to the test done locally)

Comment (Shubhranshu153) [reply cmd/finch/main.go:36]: Added this test: https://github.com/runfinch/common-tests/pull/180
Once that is merged would rebase it.

**Review by swagatbora90 — DISMISSED** [review]

**Review by pendo324 — APPROVED** [review]

**Review by swagatbora90 — APPROVED** [review]

**Pull request merged**